packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -24,7 +32,28 @@ import type {
 	NavigatorProviderProps,
 	NavigatorLocation,
 	NavigatorContext as NavigatorContextType,
+	NavigatorScreen as NavigatorScreenType,
 } from '../types';
+import { patternMatch } from '../utils/router';
+
+type MatchedPath = { params: object; id: string } | false;
```

Comment:
We can DRY this type by deriving it from the `patternMatch` function

```suggestion
type MatchedPath = ReturnType< typeof patternMatch >;
```